src/test/java/ru/ayaz/UserTest.java

```diff
@@ -0,0 +1,54 @@
+package ru.ayaz;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by ayaz on 11/5/16.
+ */
+public class UserTest {
+
+    private User user;
+
+    @Before
+    public final void before(){ user = new User(); }
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void passingNull_setName() throws Exception {
+        exception.expect(IllegalArgumentException.class);
+        user.setNickName(null);
+    }
+
+    @Test
+    public void passingEmpty_setName() throws  Exception {
+        exception.expect(IllegalArgumentException.class);
```

Comment:
Well, yeah, I looked a bit closer on ExpectedException, have a look yourself on [expecting exceptions in Junit](http://baddotrobot.com/blog/2012/03/27/expecting-exception-with-junit-rule/index.html).
1) You shouldn't return null to express that exception happened, throwing exception is better.
2) In this case (and many others) you can successfully get away with using annotation to expect exception of a certain class. IllegalArgumentException can't be thrown from anywhere else that setNickName cause its the only method call in this test. There shouldn't be any problem with that. If you are afraid that your constructor new User() in @Before method might throw IllegalArgumentException I don't think Junit will let that slide - check that yourself (try throwing exception in User constructor and see what happens) I'd like to know what happens too ;) 
3) In case you want to check the exact message your exception has then you should use the construction with ExpectedException @Rule like you do here, otherwise its excessively cumbersome.